### ISSUE TITLE
BUG: Missing removeEventListener when setupGlobals 

### ISSUE THREAD

**Issue by vuptt** [daily-co/react-native-daily-js#57]
<img width="660" alt="Screenshot 2024-09-10 at 1 53 29 PM" src="https://github.com/user-attachments/assets/e985ac1c-cc80-4027-ade0-31556f6ef37a">

This code can cause another library crash when they try to use the global variables. Example: @amityco/ts-sdk-react-native

https://github.com/daily-co/react-native-daily-js/blob/react-native-daily-js-releases/src/index.ts#L101

**Comment by filipi87** [comment]
Hi @vuptt, thanks for reporting this. 

Could you explain how this code might cause another library to crash ? Also, If you can share some code snippets that we can use to reproduce the issue it would be great.

**Comment by vuptt** [comment]
file: `node_modules/@amityco/ts-sdk-react-native/src/client/utils/onOffline.ts`

```typescript
export const onOffline = (callback: () => void) => {
  if (typeof window !== 'undefined' && window.addEventListener) {
    window.addEventListener('offline', callback);
    return () => window.removeEventListener('offline', callback);
  }

  if (typeof document !== 'undefined' && document.addEventListener) {
    document.addEventListener('offline', callback);
    return () => document.removeEventListener('offline', callback);
  }

  //  @TODO: Please update code below to follow our coding conventions and our TS rules
  if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    let unsubscribeFn = () => {};

    import('@react-native-community/netinfo').then(NetInfo => {
      unsubscribeFn = NetInfo.addEventListener(state => {
        if (state.isConnected) return;
        callback();
      });
    });

    return () => {
      unsubscribeFn();
    };
  }

  // Handle unsupported environment
  console.error('Unsupported environment');
  return () => console.error('Unsupported environment');
};
```

**Comment by filipi87** [comment]
Hi @vuptt, Just to give you an update, we plan to fix this before our next release. I’ll let you know once it’s ready.

**Comment by filipi87** [comment]
This issue has been fixed in `react-native-daily-js` version [0.70.0](https://github.com/daily-co/react-native-daily-js/releases/tag/react-native-daily-js-2024-10-31-0.70.0).
